compiler/z/codegen/BinaryEvaluator.cpp

```diff
@@ -1898,7 +1949,52 @@ OMR::Z::TreeEvaluator::tryToReplaceLongAndWithRotateInstruction(TR::Node * node,
                // this instruction sets the rotation factor to 0 and sets the zero bit(0x80).
                // So it's effectively zeroing out every bit except the inclusive range of lsBit to msBit
                // The bits in that range are preserved as is
-               generateRIEInstruction(cg, opCode, node, targetReg, sourceReg, msBit, 0x80 + lsBit, 0);
+               int32_t rangeStart = msBit;
+               int32_t rangeEnd = lsBit;
+
+               if (shiftAmount <= 0)
+                  {
+                  if (msBit - shiftAmount > 63)
+                     {
+                     return NULL;
```

Comment:
These checks should happen before we set `doTransformation = true` because we've already printed in the log that we are about to perform the transformation.

Reply:
Now that we've moved these checks above these ones can be removed.